runtime/lua/vim/diagnostic.lua

```diff
@@ -228,6 +229,12 @@ end
 --- See |nvim_buf_set_extmark()|.
 --- @field virt_text_hide? boolean
 
+--- @class vim.diagnostic.Opts.VirtualLines
+---
+--- Only show diagnostics for the current line.
+--- (default: `true`)
```

Comment:
I think the current interface is more akin to inlay hints, so default `false` makes more sense.

(If and how this is wired up to the "show diagnostic" and "next diagnostic" functions is a different -- follow-up -- question.)